[FAB-17431] Decouple javaenv from Fabric version

denyeart opened the pull request:
#### Type of change

- Improvement

#### Description

By default Fabric was configured to use the following javaenv image for java chaincode:
`runtime: $(DOCKER_NS)/fabric-javaenv:$(ARCH)-$(PROJECT_VERSION)`

Since javaenv versioning may deviate from Fabric versioning,
it is important to decouple the versions. This commit updates
the javaenv reference to use the latest two digit version:
`runtime: $(DOCKER_NS)/fabric-javaenv:$(TWO_DIGIT_VERSION)`

For example, Fabric v1.4.6 would utilize fabric-javaenv:1.4.

This is a backport of the two digit change in master, applied to javaenv.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

#### Related issues

https://jira.hyperledger.org/browse/FAB-17431
https://github.com/hyperledger/fabric/pull/555

muralisrini commented:
Let me test and unblock given this is a unit test issue.

muralisrini commented:
I'm still seeing a panic from #728 but I see there;s a PR for that out there. This problem appears to have gone away.

denyeart committed:
[FAB-17431] Decouple javaenv from Fabric version

By default Fabric was configured to use the following javaenv image for java chaincode:
runtime: $(DOCKER_NS)/fabric-javaenv:$(ARCH)-$(PROJECT_VERSION)

Since javaenv versioning may deviate from Fabric versioning,
it is important to decouple the versions. This commit updates
the javaenv reference to use the latest two digit version:
runtime: $(DOCKER_NS)/fabric-javaenv:$(TWO_DIGIT_VERSION)

For example, Fabric v1.4.5 would utilize fabric-javaenv:1.4.

This is a backport of the two digit change in master, applied to javaenv.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>
